tensorflow_addons/metrics/r_square.py

```diff
@@ -180,7 +181,7 @@ def result(self) -> tf.Tensor:
 
     def reset_states(self) -> None:
         # The state of the metric will be reset at the start of each epoch.
-        K.batch_set_value([(v, tf.zeros_like(v)) for v in self.variables])
+        K.batch_set_value([(v, np.zeros(v.shape)) for v in self.variables])
```

Comment:
How about
```
for v in self.variables:
  v.assign(tf.zeros_like(v))
```
? Is there any reason to use batch_set_value

Reply:
I think the suggected code definitely work. 

I'm not the owner, but can guess that `batch_set_value` is compatible with TF1. [code](https://github.com/keras-team/keras/blob/d71247dcd805e58110a784b03cf2fcbaa1c837c8/keras/backend.py#L3770).

Also `batch_set_value` is widely used in kinds of metrics of keras. [code](https://github.com/keras-team/keras/blob/d71247dcd805e58110a784b03cf2fcbaa1c837c8/keras/metrics.py).

Reply:
Hi fsx950233, what do you think about `batch_set_value`? Very appreciate if you can go on with reviewing even approval